### DISCUSSION TITLE
Logfiles when running as a Windows service

### DISCUSSION THREAD

**Discussion by xgcssch** [vectordotdev/vector#6067] [Q&A] [answered]
vector 0.11.1 (v0.11.1 x86_64-pc-windows-msvc 2020-12-17)

Hi!

Is there a way to get the output of vector when running as a Windows service?
Otherwise it is a little bit like wandering in the dark.

Cheers 
Sönke

**Comment by jszwedko** [comment]
This I'm not sure about. Maybe @oktal has thoughts? They introduced the windows service support in https://github.com/timberio/vector/pull/2896

**Comment by JeanMertz** [comment, marked as answer]
It does appear that any logs sent to `stdout` are effectively ending up in `/dev/null` when running a program as a Windows service (ref: https://stackoverflow.com/questions/20523480/windows-service-output).

Having said that, whatever is sent to stdout is also accessible as a regular `internal_events` source in Vector, so you can hook into that source, and then use the `file` sink to store the logs in a file, or use any of the other sinks to route the internal Vector events to your desired destination.
